### PR TITLE
Avoid compiling in parallel same time without that

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -192,16 +192,8 @@ jobs:
             /home/runner/go/pkg/mod
             /home/runner/.cache/go-build
           key: go-pkg-mod-${{ runner.os }}-${{ hashFiles('**/go.mod', '**/go.sum') }}
-      - name: Cache Tools
```
Comment (jrcamp) [inline .github/workflows/build-and-test.yml:195]: Why removed?

File: .github/workflows/build-and-test.yml [hunk 2]
```diff
@@ -192,16 +192,8 @@ jobs:
...
-        id: tool-cache
-        uses: actions/cache@v2
-        env:
-          cache-name: cache-tool-binaries
-        with:
-          path: /home/runner/go/bin
-          key: tools-${{ runner.os }}-${{ hashFiles('./internal/tools/go.mod', './cmd/mdatagen/go.mod', './cmd/mdatagen/*.go') }}
       - name: Build Collector for All Architectures
-        run: make -j4 binaries-all-sys
```
Comment (jrcamp) [inline .github/workflows/build-and-test.yml:204]: What was the issue? Using too much memory since the compiler also runs in parallel?

Comment (bogdandrutu) [reply .github/workflows/build-and-test.yml:204]: Yes, machine too small to be able to compile in parallel

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (bogdandrutu) [reply .github/workflows/build-and-test.yml:195]: Not needed.

Comment (jrcamp) [reply .github/workflows/build-and-test.yml:195]: mdatagen is used by build but probably not worth the cost of restoring cache just for that.

**Pull request merged**